terraform/modules/shared/main.tf

```diff
@@ -457,6 +457,26 @@ resource "azurerm_container_app_environment_storage" "tabulation_storage" {
   access_mode                  = "ReadWrite"
 }
 
+#### Kubernetes Service ####
+
+resource "azurerm_kubernetes_cluster" "cluster" {
+  name                = "phdi-${terraform.workspace}-cluster"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  dns_prefix          = "phdi-${terraform.workspace}"
+
+  default_node_pool {
+    name       = "default"
+    node_count = 1
+    vm_size    = "Standard_D2_v2"
+  }
+
+  identity {
+    type         = "UserAssigned"
+    identity_ids = [azurerm_user_assigned_identity.pipeline_runner.id]
+  }
```

Comment:
I think we might want to give the cluster its own managed identity. From a security/access stand point there are probably things that the cluster should have permission do that the pipeline should not (possibly vice versa as well). For example, the cluster should read and write from blob storage, but the pipeline only ever calls services in the cluster. I do realize we do have this separation of concerns between the pipeline and the container apps though.